encode/vp8enc.c

```diff
@@ -0,0 +1,1074 @@
+
+    mmap_start = frame_start_pos & (~0xfff);
+    mmap_size = (settings.frame_size + (frame_start_pos & 0xfff) + 0xfff) & (~0xfff);
+    yuv_mmap_ptr = mmap(0, mmap_size, PROT_READ, MAP_SHARED,
+                fileno(yuv_fp), mmap_start);
+
+    if (yuv_mmap_ptr == MAP_FAILED) {
+      fprintf(stderr,"Error: Failed to mmap YUV file.\n");
+      assert(0);
+    }
+
+    y_src = (uint8_t*)yuv_mmap_ptr+(frame_start_pos & 0xfff);
+    u_src = y_src + y_size; /* UV offset for NV12 */
+    v_src = y_src + y_size + u_size;
+
+
+    va_status = vaDeriveImage(vaapi_context.display, surface_id, &surface_image);
+    CHECK_VASTATUS(va_status,"vaDeriveImage");
+
+    vaMapBuffer(vaapi_context.display, surface_image.buf, &surface_p);
+    assert(VA_STATUS_SUCCESS == va_status);
+
+
+    y_dst = surface_p + surface_image.offsets[0];
+    u_dst = surface_p + surface_image.offsets[1]; /* UV offset for NV12 */
+    v_dst = surface_p + surface_image.offsets[2];
+
+    /* Y plane */
+    for (row = 0; row < surface_image.height; row++) {
+        memcpy(y_dst, y_src, surface_image.width);
+        y_dst += surface_image.pitches[0];
+        y_src += settings.width;
+    }
+
+    if (surface_image.format.fourcc == VA_FOURCC_NV12) { /* UV plane */
+        for (row = 0; row < surface_image.height / 2; row++) {
+            for (col = 0; col < surface_image.width / 2; col++) {
+                u_dst[col * 2] = u_src[col];
+                u_dst[col * 2 + 1] = v_src[col];
+            }
+
+            u_dst += surface_image.pitches[1];
+            u_src += (settings.width / 2);
+            v_src += (settings.width / 2);
+        }
+    } else if (surface_image.format.fourcc == VA_FOURCC_YV12 ||
+               surface_image.format.fourcc == VA_FOURCC_I420) {
+        const int U = surface_image.format.fourcc == VA_FOURCC_I420 ? 1 : 2;
+        const int V = surface_image.format.fourcc == VA_FOURCC_I420 ? 2 : 1;
+
+        u_dst = surface_p + surface_image.offsets[U];
+        v_dst = surface_p + surface_image.offsets[V];
+
+        for (row = 0; row < surface_image.height / 2; row++) {
+            memcpy(u_dst, u_src, surface_image.width / 2);
+            memcpy(v_dst, v_src, surface_image.width / 2);
+            u_dst += surface_image.pitches[U];
+            v_dst += surface_image.pitches[V];
+            u_src += (settings.width / 2);
+            v_src += (settings.width / 2);
+        }
+    }
+
+    vaUnmapBuffer(vaapi_context.display, surface_image.buf);
+    vaDestroyImage(vaapi_context.display, surface_image.image_id);
+
+    if (yuv_mmap_ptr)
+      munmap(yuv_mmap_ptr, mmap_size);
+}
+/********************************************
+*
+* END: Read YUV Input File Releated Stuff
+*
+********************************************/
+
+void vp8enc_init_QMatrix(VAQMatrixBufferVP8 *qMatrix)
+{
+
+  for (size_t i = 0; i < N_ELEMENTS(qMatrix->quantization_index); i++) {
+      // What is the quantization matrix exactly???
```

Comment:
Check the comment of VAQMatrixBufferVP8 in va_enc_vp8.h.
default quat values are provided in the vp8 spec. Please remove this comment.
Since we are not enabling segmentation, only quantization_index[0] will be used.